autotest/gdrivers/jp2grok.py

```diff
@@ -0,0 +1,3013 @@
+#!/usr/bin/env pytest
+# -*- coding: utf-8 -*-
+###############################################################################
+# $Id$
+#
+# Project:  GDAL/OGR Test Suite
+# Purpose:  Test read/write functionality for JP2Grok driver.
+# Author:   Even Rouault <even dot rouault at spatialys.com>,
+#           Frank Warmerdam <warmerdam@pobox.com>
+#
+###############################################################################
+# Copyright (c) 2010-2021, Even Rouault <even dot rouault at spatialys.com>
+#
+# Permission is hereby granted, free of charge, to any person obtaining a
+# copy of this software and associated documentation files (the "Software"),
+# to deal in the Software without restriction, including without limitation
+# the rights to use, copy, modify, merge, publish, distribute, sublicense,
+# and/or sell copies of the Software, and to permit persons to whom the
+# Software is furnished to do so, subject to the following conditions:
+#
+# The above copyright notice and this permission notice shall be included
+# in all copies or substantial portions of the Software.
+#
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS
+# OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL
+# THE AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING
+# FROM, OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER
+# DEALINGS IN THE SOFTWARE.
+###############################################################################
+
+import os
+import sys
+import shutil
+from osgeo import gdal
+from osgeo import ogr
+from osgeo import osr
+import pytest
+
+import gdaltest
+
+pytestmark = pytest.mark.require_driver('JP2Grok')
+
```

Comment:
As I raised previously in https://github.com/OSGeo/gdal/pull/5117/files#r793891898 , there's a lot of similarity between that file and jp2openjpeg.py  .  I presume using pytest fixtures we could factor a lot and most differences are perhaps in checksums for lossy compression (I didn't check) . Any opinion @dbaston if that's worth it ?    The potential downsides would be if the behavior of drivers would have to diverge in significant ways, but I don't anticipate that

Reply:
@rouault any ideas on how we get to a common test base for both drivers ?  Also, can I run the python driver test locally ?

Reply:
> any ideas on how we get to a common test base for both drivers ?

presumably using pytest.fixture() in test_jp2openjpeg.py.  The driver_name fixture in autotest/ogr/ogr_rfc35py (recent master) could serve as inspiration. Presumably something along (untested...):

```
@pytest.fixture(autouse=True, params=["JP2OpenJPEG", "JP2Grok"])
def driver(request):
    driver_name = request.param
    jp2_drv = gdal.GetDriverByName(driver_name)
    if jp2_drv is None:
        pytest.skip(f"{driver_name} not available")
    gdaltest.deregister_all_jpeg2000_drivers_but(driver_name)
    yield jp2_drv
    gdaltest.reregister_all_jpeg2000_drivers()

def test_1(driver):
     ... do something ....

def test_2(driver):
     ... do something ....
```

> Also, can I run the python driver test locally ?

Sure, you need to have the swig and python-dev packages installed. Then:
- cmake -DBUILD_PYTHON_BINDINGS=ON ..
- make python_binding
- source ../scripts/setenv.sh
- pytest autotest/gdrivers/jp2openjpeg.py

Reply:
Thanks.